protostar/commands/init/project_creator/new_project_creator.py

```diff
@@ -120,8 +120,10 @@ def _new_project_config(
 
     @staticmethod
     def _validate_project_name(name: str):
-        pattern = r"^[a-zA-Z0-9][a-zA-Z0-9_]*$"
-        if not re.match(pattern, name):
+        # https://github.com/software-mansion/scarb/blob/main/scarb/src/core/package/name.rs#LL42C9
+        pattern = r"^[a-zA-Z_][0-9a-zA-Z_]*$"
+        if not re.match(pattern, name) or name == "_":
             raise ProtostarException(
-                f"Provided project name {name} does not match regex {pattern}. Choose a different project name."
+                f"Provided project name {name} does not match the regex {pattern} or is equal to '_'. "
+                f"Choose a different project name."
             )
```

Comment:
It's clearer with the simplified regex. How we can improve it is I think we can have separate error messages for `_` and rest of the pattern.

Reply:
I would leave it as it is, I guess a user shouldn't have problems recognising what part of the error message applies to his name.

Reply:
They shouldn't but handling these cases separately doesn't require much work and our side and provides a slightly better UX I think.

Reply:
I think it will only introduce more cognitive complexity, it doesn't provides any significant UX improvement.

Reply:
Okay, let's keep it as is.